readtime.py

```diff
@@ -1,99 +1,159 @@
-        None
-    """
-    if content_type_supported(content):
-        # We get the content's text, split it at the spaces and check the
-        # length of the provided array to get a good estimation on the amount
-        # of words in the content.
-        words = len(content.content.split())
-        read_time_seconds = round((words / AVERAGE_READING_WPM) * 60, 2)
-        minutes, seconds = get_time_from_seconds(read_time_seconds)
-        minutes_str = pluralize(minutes, "Minute", "Minutes")
-        seconds_str = pluralize(seconds, "Second", "Seconds")
-        content.readtime = "{} and {}".format(minutes_str, seconds_str)
-        content.readtime_minutes = minutes + int(bool(seconds))
+        try:
 
+            self.initialized = True
 
-def pluralize(measure, singular, plural):
-    """ Returns a string that contains the measure (amount) and its plural or
-    singular form depending on the amount.
+            settings_contenttype = sender.settings.get('READTIME_CONTENT_SUPPORT', self.content_type_supported)
 
-    Parameters:
-        :param measure: Amount, value, always a numerical value
-        :param singular: The singular form of the chosen word
-        :param plural: The plural form of the chosen word
+            if not isinstance(settings_contenttype, list):
+                raise Exception("Settings 'READTIME_CONTENT_SUPPORT' is not a list()")
+            else:
+                self.content_type_supported = settings_contenttype
 
-    Returns:
-        String
-    """
-    if measure == 1:
-        return "{} {}".format(measure, singular)
-    else:
-        return "{} {}".format(measure, plural)
+            settings_wpm = sender.settings.get('READTIME_LANGUAGE_SUPPORT', self.language_settings)
 
+            # Allows a wpm entry
+            if isinstance(settings_wpm, int):
+                self.language_settings['default']['wpm'] = settings_wpm
 
-def get_time_from_seconds(read_time_seconds):
-    """ Returns, in a tuple, the amount of minutes and seconds based on the
-    provided seconds.  87 seconds would return (1, 27); 1 minute and 27 seconds.
+            # Default checker
+            elif isinstance(settings_wpm, dict):
+                if 'default' not in settings_wpm:
+                    pass
+                else:
 
-    Parameters:
-        :param read_time_seconds: The amount of seconds to retrieve time from.
+                    for key in settings_wpm.keys():
 
-    Returns:
-        Tuple containing minutes and seconds
-    """
-    minutes = int(read_time_seconds / SECONDS_IN_MINUTE)
-    seconds = int(read_time_seconds % SECONDS_IN_MINUTE)
-    return minutes, seconds
+                        if "wpm" not in settings_wpm[key]:
+                            raise Exception("Missing wpm value for the language: %s" % key)
 
+                        if not isinstance(settings_wpm[key]['wpm'], int):
+                            raise Exception("WPM is not an integer for the language: %s" % key)
```

Comment:
There's a mix of two python string formatting `%s` vs `'{}'.format()`. Maybe one would be better?